response_operations_ui/views/respondents.py

```diff
@@ -14,22 +18,93 @@
                           static_folder='static', template_folder='templates')
 
 
-@respondent_bp.route('/', methods=['GET', 'POST'])
+@respondent_bp.route('/', methods=['GET'])
+@login_required
+def respondent_home():
+    return render_template('respondent-search/search-respondents.html',
+                           form=RespondentSearchForm(),
+                           breadcrumbs=[{"title": "Respondents"}])
+
+
+@respondent_bp.route('/search', methods=['POST'])
+@login_required
+def search_redirect():
+    form = RespondentSearchForm()
+    form_valid = form.validate()
+
+    if not form_valid:
+        flash('At least one input should be filled')
+
+    source = form.source.data or 'home'
+    page = request.args.get('page', 1)
+
+    query_string = urlencode({
+        'email_address': form.email_address.data or '',
+        'first_name': form.first_name.data or '',
+        'last_name': form.last_name.data or '',
+        'source': source,
+        'page': page
+    })
+
+    if not form_valid:
+        redirect_url = url_for('respondent_bp.respondent_home')
+    else:
+        redirect_url = urljoin(url_for('respondent_bp.respondent_search'), '?' + query_string)
+
+    return redirect(redirect_url)
+
+
+@respondent_bp.route('/search')
+@respondent_bp.route('/search/', methods=['GET'])
 @login_required
 def respondent_search():
-    form = SearchForm()
-    breadcrumbs = [{"title": "Respondents"}]
-    response = None
+    breadcrumbs = [{"title": "Respondents"}, {"title": "Search"}]
+
+    args = get_controller_args_from_request(request)
+
+    first_name = args['first_name']
+    last_name = args['last_name']
+    email_address = args['email_address']
+    page = args['page']
+
+    form = RespondentSearchForm()
+
+    form.first_name.data = first_name
+    form.last_name.data = last_name
+    form.email_address.data = email_address
+
+    party_response = party_controller.search_respondents(first_name, last_name, email_address, page)
+
+    respondents = party_response.get('data', [])
+    total_respondents_available = party_response.get('total', 0)
+
+    filtered_respondents = filter_respondents(respondents)
+
+    RESULTS_PER_PAGE = app.config["PARTY_RESPONDENTS_PER_PAGE"]
+
+    offset = (int(page) - 1) * RESULTS_PER_PAGE
+
+    first_index = 1 + offset
```

Comment:
You're only using both of these variables once, might as well inline them in the render_template below

Reply:
Removed in next push